### PR TITLE
[minor]  Use root NS specification for opcoded functions

### PR DESCRIPTION
**Pull request by kalessil**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | n/a

The optimization (the functions have opcodes) is effective for PHP 7.0+

### CODE REVIEW

**Comment by kalessil** [comment]
@samdark : here we go with the 2nd PR, It would be great to get an idea how it affects a running system.

**Comment by SilverFire** [comment]
Have you measured performance influence somehow?

File: build/controllers/DevController.php [hunk 1]
```diff
@@ -98,17 +98,17 @@ public function actionAll()
...
-        $base = dirname(dirname(__DIR__));
+        $base = \dirname(\dirname(__DIR__));
         $dirs = $this->listSubDirs("$base/extensions");
         $dirs = array_merge($dirs, $this->listSubDirs("$base/apps"));
         asort($dirs);
 
         $oldcwd = getcwd();
         foreach ($dirs as $dir) {
-            $displayDir = substr($dir, strlen($base));
+            $displayDir = substr($dir, \strlen($base));
```
Comment (samdark) [inline build/controllers/DevController.php]: Why not `substr`?

File: build/controllers/ClassmapController.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ public function actionCreate($root = null, $mapFile = null)
             if (strpos($file, $root) !== 0) {
                 throw new Exception("Something wrong: $file\n");
             }
-            $path = str_replace('\\', '/', substr($file, strlen($root)));
+            $path = str_replace('\\', '/', substr($file, \strlen($root)));
```
Comment (samdark) [inline build/controllers/ClassmapController.php]: Why not `substr` and `str_replace`?

File: build/controllers/PhpDocController.php [hunk 3]
```diff
@@ -195,7 +195,7 @@ protected function findFiles($root, $needsInclude = true)
 //                return [];
 //            }
         } elseif (preg_match('~apps/([\w-]+)[\\\\/]?$~', $root, $matches)) {
-            $extensionPath = dirname(dirname(rtrim($root, '\\/'))) . '/extensions';
+            $extensionPath = \dirname(\dirname(rtrim($root, '\\/'))) . '/extensions';
```
Comment (samdark) [inline build/controllers/PhpDocController.php]: Why not `rtrim`?

File: build/controllers/ClassmapController.php (hunk 2 above)
Comment (kalessil) [reply build/controllers/ClassmapController.php]: They don't have opcodes, the PR addresses only functions which gets opcodes generated (hence can contribute to better performance). Here the reference which functions covered: https://github.com/php/php-src/blob/f2db305fa4e9bd7d04d567822687ec714aedcdb5/Zend/zend_compile.c#L3872

**Comment by kalessil** [comment]
Nope: ATM I don't have a running system where I can give it a try and capture numbers.

**Comment by samdark** [comment]
Tests are fine and I don't see any harm in having it.

**Comment by samdark** [comment]
Merged. Thank you!

**Comment by kalessil** [comment]
Thank you @samdark 🎉

**Pull request merged**